e2e/support/commands/api/composite/createNativeQuestionAndDashboard.js

```diff
@@ -5,14 +5,22 @@ Cypress.Commands.add(
       ({ body: { id: questionId } }) => {
         cy.createDashboard(dashboardDetails).then(
           ({ body: { id: dashboardId } }) => {
-            cy.request("POST", `/api/dashboard/${dashboardId}/cards`, {
-              cardId: questionId,
-              // Add sane defaults for the dashboard card size and position
-              row: 0,
-              col: 0,
-              size_x: 8,
-              size_y: 6,
-            });
+            cy.request("PUT", `/api/dashboard/${dashboardId}/cards`, {
+              cards: [
+                {
+                  id: -1,
```

Comment:
In [`dashboard/actions/data-fetching.js`](https://github.com/metabase/metabase/pull/29641/files#diff-636074b28010ac8c8f8597b27dc79a72530c430dbfd344340c399de22a1f6c80R82), the id for new dashcards used to be a random float between 0 and 1. In this PR we changed it to be negative numbers.

For new dashcards that the user has added while in edit mode, we need to have a unique id for each one so the client can differentiate them. This is why we can't use a single value like `null`, or a static descriptor like `"new"`. We also don't want to change the type of this value from a `number` to something else like a `string`, because doing so would break any typescript code that expects the dashcard id to be a `number`.

I agree it's not intuitive, but I don't think this is an issue. We don't ever manually construct and send the api request to add new dashcards, it's always done by dispatching a [`ADD_CARD_TO_DASH` redux action](https://github.com/metabase/metabase/blob/92a094159958a81ff0076058c826dbeb8cc3d011/frontend/src/metabase/dashboard/actions/cards.js#L55), which handles setting the id for you. 

In e2e I've refactored everything to use one of the helper methods that sends the request. I supposed we could remove it from the e2e code, but I'd rather keep it the same as how it's being done in the client, especially since the API expects that value to be populated.